core/src/main/java/pl/netigen/core/ads/AdsManager.kt

```diff
@@ -125,11 +132,24 @@ class AdsManager(var viewModel: NetigenViewModel, val activity: AppCompatActivit
 
     fun isOnline(): Boolean {
         val cm = activity.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
-        var netInfo: NetworkInfo? = null
-        if (cm != null) {
-            netInfo = cm.activeNetworkInfo
-        }
+        val netInfo: NetworkInfo? = cm.activeNetworkInfo
         return netInfo != null && netInfo.isConnectedOrConnecting
     }
 
+    fun onDestroy() {
+        rewardedAdManager?.onDestroy()
+    }
+
+    fun splashScreenOnStop() {
+        viewModel.isSplashInBackground = true
+    }
+
+    fun splashScreenOnStart() {
+        viewModel.isSplashInBackground = false
+    }
+
+    fun splashScreenOnDestroy() {
+        interstitialAdManager.interstitialAdHandler.removeCallbacksAndMessages(null)
```

Comment:
Here should be rather interstitialAdManager.destroy() or onDestroySplash() , interstitialAdManager.interstitialAdHandler shoudl not be public

Reply:
Done